src/thunderbots/software/ai/hl/stp/stp.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <random>
+
+#include "ai/hl/hl.h"
+#include "ai/hl/stp/play/play.h"
+#include "ai/intent/intent.h"
+
+/**
+ * The STP module is an implementation of the high-level logic Abstract class, that
+ * uses the STP (Skills, Tactics, Plays) framework for its decision making.
+ */
+class STP : public HL
+{
+   public:
+    /**
+     * Creates a new High-Level logic module that uses the STP framework for
+     * decision-making.
+     *
+     * @param random_seed The random seed used for STP's internal random number generator.
+     * Defaults to the number of nanoseconds since the Linux epoch
+     */
+    explicit STP(
+        long random_seed = std::chrono::system_clock::now().time_since_epoch().count());
```

Comment:
I think `1` would be a better default value here, as the time is not deterministic

Reply:
The point is for the default value to be non-deterministic. Random seeds are required for "true" randomness. We allow the seed to be specified manually so tests can be deterministic.

Reply:
Unless you don't want any default value, and force the caller to be explicit about what seed it wants?

Reply:
I removed the default value to the caller has to be explicit about the seed.

Reply:
Nevermind, I changed it back since not having a default constructor made tests difficult. However I've made the default value `0` instead so callers have to opt-in to "full randomness"